source/elements/oneDAL/include/oneapi/dal/array.hpp

```diff
@@ -1,276 +1,331 @@
 namespace oneapi::dal {
 
 /// @tparam Data The type of the memory block elements within the array.
-///              $Data$ can represent any type.
+///              ``Data`` can represent any :capterm:`data type`.
 ///
-/// @pre $Data$ cannot be const-qualified.
+/// @pre ``Data`` cannot be const-qualified.
 template <typename Data>
 class array {
 public:
     using data_t = Data;
 
-public:
-    /// Allocates a new memory block for mutable data, does not initialize it,
-    /// creates a new array instance by passing a pointer to the memory block.
-    /// The array shall own the memory block (for details, see :txtref:`data_ownership_requirements`).
+    /// Creates a new array instance by allocating a mutable memory block. The created array manages
+    /// lifetime of the allocated memory block.
```

Comment:
```suggestion
    /// the lifetime of the allocated memory block.
```

same change in other places